src/shared/infra/repositories/user_repository_mock.ts

```diff
@@ -72,4 +80,11 @@ export class UserRepositoryMock implements IUserRepository {
   async getUserCounter(): Promise<number> {
     return this.userCounter
   }
+  async loginUser(email: string, password: string): Promise<User> {
+    const user = this.users.find(user => user.email === email && user.password === password)
+    if (!user) {
+      throw new NoItemsFound('Invalid email or password')
```

Comment:
This NoItemsFound should be throw on the usecase, not the repo, if it can't find the user returns Null